ATSketchKit/ATSketchView+UndoRedo.swift

```diff
@@ -32,16 +32,21 @@ extension ATSketchView {
 			mostRecentLayer!.removeFromSuperlayer()
 		}
 		self.setNeedsDisplay()
+    self.delegate?.sketchViewUndoRedoUpdated(self)
 	}
 	
 	public func redo() {
 		let mostRecentUndoLayer = self.history.last
 		
 		if mostRecentUndoLayer != nil {
-			self.layer.addSublayer(mostRecentUndoLayer!)
+      self.layer.insertSublayer(mostRecentUndoLayer!, below: self.topLayer)
 			self.history.removeLast()
 		}
 		self.setNeedsDisplay()
+    self.delegate?.sketchViewUndoRedoUpdated(self)
 	}
-	
+
+  public func clearRedo() {
```

Comment:
I would prefer flushHistory as a name for the function. It's more explicit of what the func actually does.